crates/polars-core/src/series/mod.rs

```diff
@@ -157,17 +157,18 @@ impl Series {
     }
 
     pub fn clear(&self) -> Series {
-        // Only the inner of objects know their type, so use this hack.
-        #[cfg(feature = "object")]
-        if matches!(self.dtype(), DataType::Object(_, _)) {
-            return if self.is_empty() {
-                self.clone()
-            } else {
-                let av = self.get(0).unwrap();
-                Series::new(self.name(), [av]).slice(0, 0)
-            };
+        match self.dtype() {
+            #[cfg(feature = "object")]
+            DataType::Object(_, _) => {
+                if self.is_empty() {
+                    self.clone()
+                } else {
+                    // SAFETY: we can always take zero elements
+                    unsafe { self.take_slice_unchecked(&[]).clone() }
```

Comment:
This works. Though this is not worth doing unsafe. Can we use `self.take`.

I also wonder why we need to `clone` afterwards? :thinking:

Reply:
Can do. Out of curiosity, is this path a bit simpler/faster than the standard dtype path of going through `full_null` with zero elements? We could just remove all the if logic and just return an empty slice for all dtypes.

Reply:
Ahh unfortunately that breaks `pl.Array`. I'll keep the special object logic in there.